Bottom: 70348eb2582e9b9ec10ed6b66dfe76ef2f568c1e
Top:    ec820c10a02d57b8b5cb01ba3cb691c0b48afffd
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-29 02:54:00 +0100

Refresh of wip-on-quad-edge-play

---

diff --git a/src/hid/gtk/appearance.c b/src/hid/gtk/appearance.c
new file mode 100644
index 0000000..92790fcc
--- /dev/null
+++ b/src/hid/gtk/appearance.c
@@ -0,0 +1,25 @@
+#include <glib.h>
+
+#include "appearance.h"
+
+appearance
+*make_appearance (void)
+{
+  appearance *appear = g_new0 (appearance, 1);
+
+  return appear;
+}
+
+void
+destroy_appearance (appearance *appear)
+{
+  g_free (appear);
+}
+
+void
+appearance_set_color (appearance *appear, float r, float g, float b)
+{
+  appear->r = r;
+  appear->g = g;
+  appear->b = b;
+}
diff --git a/src/hid/gtk/appearance.h b/src/hid/gtk/appearance.h
new file mode 100644
index 0000000..d4917d7
--- /dev/null
+++ b/src/hid/gtk/appearance.h
@@ -0,0 +1,7 @@
+typedef struct {
+  float r, g, b;
+} appearance;
+
+appearance *make_appearance (void);
+void destroy_appearance (appearance *appearance);
+void appearance_set_color (appearance *appearance, float r, float g, float b);
